tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,322 @@
+        name="SJER/Reflectance/Metadata/Spectral_Data/Wavelength",
+        write.attributes=T)
+```
+
+## Spatial Subsetting
+
+Even after spectral subsetting, our file size would be greater than 100Mb. Therefore, we will also perform a *spatial* subsetting processto further reduce our file size. Now, we need to figure out which part of the full image that we want to extract for our subset. It takes a number of steps in order to read in a band of data and plot the reflectance values - all of which are thoroughly described in the <a href="{{ site.baseurl }}/hsi-hdf5-r" target="_blank"> *Intro to Working with Hyperspectral Remote Sensing Data in HDF5 Format in R*</a> tutorial. For now, let's focus on the essentials for our problem at hand. In order to explore the spatial qualities of this dataset, let's plot a single band as an overview map to see what objects and land cover types are contained within this mosaic tile. The Reflectance_Data dataset has three dimensions in the order of bands, columns, rows. We want to extract a single band, and all 1,000 columns and rows, so we will feed those values into the `index=` argument as a `list`. For this example, we will select the 58th band in the hyperspectral dataset, which corresponds to a wavelength of 667nm, which is in the red end of the visible electromagnetic spectrum. We will use `NULL` in the column and row position to indicate that we want *all* of the columns and rows (we agree that it is weird that `NULL` indicates "all" in this circumstance, but that is the default behavior for this, and many other, functions).
+
+
+```{r plot-example-band}
+# Extract or "slice" data for band 58 from the HDF5 file
+b58 <- h5read(f,name = "SJER/Reflectance/Reflectance_Data",
+             index=list(58,NULL,NULL))
+h5closeAll()
+
+# convert from array to matrix
+b58 <- b58[1,,]
+
+# Make a plot to view this band
+image(log(b58), col=grey(0:100/100))
+
+```
+
+As we can see here, this hyperspectral reflectance tile contains a school campus that is under construction. There are many different land cover types contained here, which makes it a great example! Perhaps the most unique feature shown is in the bottom right corner of this image, where we can see the tip of a small reservoir. Let's be sure to capture this feature in our spatial subset, as well as a few other land cover types (irrigated grass, trees, bare soil, and buildings).
+
+While raster images count their pixels from the top left corner, we are working with a matrix, which counts its pixels from the bottom left corner. Therefore, rows are counted from the bottom to the top, and columns are counted from the left to the right. If we want to sample the bottom right quadrant of this image, we need to select rows 1 through 500 (bottom half), and columns 501 through 1000 (right half). Note that, as above, the `index=` argument in `h5read()` requires a `list` of three dimenstions for this example - in the order of bands, columns, rows. 
+
```

Comment:
I have a much better understanding for why & how these numbers come about.  Thanks for the added details.

Reply:
Thank _you_ for the good suggestion!